lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -174,6 +174,21 @@ public void Close(byte[] message)
                 _currentStatus.SetException(TaskClientCodeException.Create(
                     TaskId, ContextId, "Error during Close().", e));
             }
+            finally
+            {
+                try
+                {
+                    if (_userTask != null)
+                    {
+                        _userTask.Dispose();
```

Comment:
This logic can also get called when the Task finishes in another thread at the same time. You will need ensure that `Dispose()` is only called once, even though it is best practice for be `Dispose()` to be reentrant.